### PR TITLE
Support params in react-native-tts speak in Android

### PR DESCRIPTION
**Pull request by sunjae1294**
- Add new argument `params` to `speak()` method

### CODE REVIEW

**Comment by sunjae1294** [comment]
@ohtangza @ghsdh3409 Please review this change!

File: android/src/main/java/net/no_mad/tts/TextToSpeechModule.java [hunk 1]
```diff
@@ -282,12 +284,54 @@ private boolean notReady(Promise promise) {
     }
 
     @SuppressWarnings("deprecation")
-    private int speak(String utterance, String utteranceId) {
+    private int speak(String utterance, String utteranceId, ReadableMap inputParams) {
+        int audioStreamType;
```
Comment (ohtangza) [inline android/src/main/java/net/no_mad/tts/TextToSpeechModule.java]: How about moving it to the line before `switch(audioStreamTypeString) {`?

File: android/src/main/java/net/no_mad/tts/TextToSpeechModule.java [hunk 2]
```diff
@@ -282,12 +284,54 @@ private boolean notReady(Promise promise) {
...
+
+        String audioStreamTypeString = inputParams.hasKey("KEY_PARAM_STREAM") ? inputParams.getString("KEY_PARAM_STREAM") : null;
+        float volume = inputParams.hasKey("KEY_PARAM_VOLUME") ? (float) inputParams.getDouble("KEY_PARAM_VOLUME") : (float) 1;
```
Comment (ohtangza) [inline android/src/main/java/net/no_mad/tts/TextToSpeechModule.java]: Minor issue. 
- `1.0f` preferable to `(float) 1` (out of curiosity)?
- I wonder wasn't there any `getFloat()` method for this.

File: index.js [hunk 3]
```diff
@@ -38,11 +38,11 @@ class Tts extends NativeEventEmitter {
     return TextToSpeech.voices();
   }
 
-  speak(utterance, voiceId) {
+  speak(utterance, params, voiceId) {
```
Comment (ohtangza) [inline index.js]: I think `params` should be added as a third parameter for the backward compatability? Any reason for this to be aded as the second parameter?

**Review by ohtangza — COMMENTED** [review]
It also requires @ghsdh3409's approval.

File: android/src/main/java/net/no_mad/tts/TextToSpeechModule.java (hunk 2 above)
Comment (sunjae1294) [reply android/src/main/java/net/no_mad/tts/TextToSpeechModule.java]: - Thanks I didn't know about `1.0f` . It looks certainly better
- Yes, there was only `getDouble` for ReadableMap type

File: index.js (hunk 3 above)
Comment (sunjae1294) [reply index.js]: If `params` is at second parameter, wouldn't `speak(utterance, {
  KEY_PARAM_STREAM: STREAM_VOICE_CALL
})` 
cause problem?

Comment (ohtangza) [reply index.js]: Hum, I agree that it seems little ugly. However, I think we'd better apply the third parameter here and use `speak(utterance, undefined, { KEY_PARAM_STREAM: STREAM_VOICE_CALL });` not to diverge from the upstream repository.

@ghsdh3409 Any opinion on this?

Comment (ghsdh3409) [reply index.js]: Yes, I agree with @ohtangza. For compatibility, `params` should be located at third argument.

File: android/src/main/java/net/no_mad/tts/TextToSpeechModule.java [hunk 4]
```diff
@@ -282,12 +284,54 @@ private boolean notReady(Promise promise) {
...
+        float pan = inputParams.hasKey("KEY_PARAM_PAN") ? (float) inputParams.getDouble("KEY_PARAM_PAN") : (float) 0;
```
Comment (ghsdh3409) [inline android/src/main/java/net/no_mad/tts/TextToSpeechModule.java]: Wow! you made also `KEY_PARAM_VOLUME` and `KEY_PARAM_PAN`!!! Very nice! Thanks!!!

Comment (ohtangza) [reply android/src/main/java/net/no_mad/tts/TextToSpeechModule.java]: I'm also very impressed... so considerate... 💯

**Comment by sunjae1294** [comment]
@ohtangza @ghsdh3409 This needs review!

**Review by ghsdh3409 — APPROVED** [review]
Really nice.

**Pull request merged**